projects/npm-tools/packages/npm-scripts/src/sass/build.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const globby = require('globby');
+const r2 = require('liferay-r2');
+const path = require('path');
+const sass = require('sass');
+
+const log = require('../utils/log');
+
+const SASS_EXCLUDE = [
+	'**/_*.scss',
+	'**/build/**',
+	'**/classes/**',
+
+	// Copy from CSSBuilderArgs.java
+
+	'**/_diffs/**',
+	'**/.sass-cache*/**',
+	'**/.sass_cache_*/**',
+	'**/_sass_cache_*/**',
+	'**/_styled/**',
+	'**/_unstyled/**',
+	'**/css/aui/**',
+	'**/css/clay/**',
+	'**/tmp/**',
+];
+
+const getClayPaths = () => {
+	try {
+		return require('@clayui/css').includePaths;
+	} catch (e) {
+		return [];
+	}
+};
+
+const clayIncludePaths = getClayPaths();
+
+async function collectSassFiles(baseDir, excludes = []) {
+	const excludePaths = [
+		...SASS_EXCLUDE.map((exclude) => path.join(baseDir, exclude)),
+		...excludes.map((exclude) => path.join(baseDir, exclude)),
+	];
+
+	return await globby([
+		path.join(baseDir, '**/*.scss'),
+		...excludePaths.map((exclude) => `!${exclude}`),
+	]);
+}
+
+function buildSass(file, output) {
+	const newFileName = file
+		.split('/')
+		[file.split('/').length - 1].replace('.scss', '.css');
+
+	const outputFile = path.join(output, newFileName);
+
+	const result = sass.renderSync({
+		file,
+		includePaths: clayIncludePaths,
```

Comment:
This should be some type of passed-in param so we can add includepaths by module.

It would definitely be great to default this to `@clayui/css` for simplicity, although that means all sass compilation times will be slower regardless of them using Clay or not.

Reply:
👍